templates/e2e-llm-workflows/README.ipynb

```diff
@@ -512,12 +491,52 @@
     }
    ],
    "source": [
+    "import anyscale\n",
+    "from ray.data.datasource import FilenameProvider\n",
+    "from rich import print as rprint\n",
+    "from typing import Dict\n",
+    "\n",
+    "import os\n",
+    "\n",
+    "class CustomFilenameProvider(FilenameProvider):\n",
+    "    def __init__(self, custom_filename: str):\n",
+    "        self.custom_filename = custom_filename\n",
+    "    \n",
+    "    def get_filename_for_block(self, *args, **kwargs) -> str:\n",
+    "        return self.custom_filename\n",
+    "\n",
     "# Write to cloud storage\n",
-    "fs = pyarrow.fs.PyFileSystem(pyarrow.fs.FSSpecHandler(S3FileSystem()))\n",
-    "storage_path = os.environ['ANYSCALE_ARTIFACT_STORAGE'][len('s3://'):]\n",
-    "ft_train_ds.write_json(f'{storage_path}/viggo/train.jsonl', filesystem=fs)\n",
-    "ft_val_ds.write_json(f'{storage_path}/viggo/val.jsonl', filesystem=fs)\n",
-    "ft_test_ds.write_json(f'{storage_path}/viggo/test.jsonl', filesystem=fs)"
+    "temp_path = '/tmp/e2e_llm/viggo'\n",
+    "os.makedirs(temp_path, exist_ok=True)\n",
+    "config_fields: Dict[str, str] = {}\n",
+    "for dataset, filename, field_name in [\n",
+    "    (ft_train_ds, 'train.jsonl', \"train_path\"),\n",
+    "    (ft_val_ds, 'val.jsonl', \"valid_path\"),\n",
+    "    (ft_test_ds, 'test.jsonl', \"test_path\")\n",
+    "]:\n",
+    "    dataset.write_json(temp_path, filename_provider=CustomFilenameProvider(filename))\n",
+    "    def get_full_name():\n",
+    "        full_name = os.environ.get(\"ANYSCALE_USERNAME\", \"default\")\n",
+    "        full_name = \"_\".join(full_name.split())  # Join the full name with underscores\n",
+    "        full_name = full_name.lower()  # Lowercase the full name\n",
+    "        return full_name\n",
+    "    \n",
+    "    # full_name/e2e_llm/viggo/train.jsonl\n",
+    "    dataset_name = f\"{get_full_name()}/e2e_llm/viggo/{filename}\"\n",
+    "    dataset = anyscale.llm.dataset.upload(\n",
+    "        f\"{temp_path}/{filename}\",\n",
+    "        dataset_name,\n",
+    "        # VIGGO dataset for E2E LLM template, 'train.jsonl' split \n",
+    "        description=(\n",
+    "            \"VIGGO dataset for E2E LLM template, \"\n",
+    "            f\"'{filename}' split\"\n",
+    "        ),\n",
+    "    )\n",
+    "    config_fields[field_name] = dataset.storage_uri\n",
+    "    rprint(f\"Metadata for '{filename}'\")\n",
+    "    rprint(dataset)\n",
+    "rprint(\"Config fields:\")\n",
+    "rprint(config_fields)"
    ]
```

Comment:
let's also update the "Load artifacts" section with the models sdk

Reply:
sg, @SumanthRH is tracking this

Reply:
Done!